app/com/arpnetworking/metrics/portal/scheduling/impl/JobScheduler.java

```diff
@@ -0,0 +1,225 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.scheduling.impl;
+
+import akka.actor.AbstractActorWithTimers;
+import akka.actor.ActorRef;
+import akka.actor.PoisonPill;
+import akka.actor.Props;
+import akka.pattern.PatternsCS;
+import com.arpnetworking.metrics.portal.scheduling.JobRef;
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import models.internal.scheduling.Job;
+import scala.Serializable;
+import scala.concurrent.duration.Duration;
+import scala.concurrent.duration.FiniteDuration;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An actor that executes {@link Job}s.
+ *
+ * @param <T> The type of result produced by the {@link Job}s.
+ *
+ * @author Spencer Pearson (spencerpearson at dropbox dot com)
+ */
+public final class JobScheduler<T> extends AbstractActorWithTimers {
+
+    private final JobRef<T> _jobRef;
+    private final Clock _clock;
+
+    /**
+     * Props factory.
+     *
+     * @param <T> The type of result produced by the {@link JobRef}'s job.
+     * @param jobRef The job to intermittently execute.
+     * @return A new props to create this actor.
+     */
+    public static <T> Props props(final JobRef<T> jobRef) {
+        return props(jobRef, Clock.systemUTC());
+    }
+
+    /**
+     * Props factory.
+     *
+     * @param <T> The type of result produced by the {@link JobRef}'s job.
+     * @param jobRef The job to intermittently execute.
+     * @param clock The clock the scheduler will use, when it ticks, to determine whether it's time to run the next job(s) yet.
+     * @return A new props to create this actor.
+     */
+    protected static <T> Props props(final JobRef<T> jobRef, final Clock clock) {
+        return Props.create(JobScheduler.class, () -> new JobScheduler<>(jobRef, clock));
+    }
+
+    private JobScheduler(final JobRef<T> jobRef, final Clock clock) {
+        _jobRef = jobRef;
+        _clock = clock;
+        timers().startPeriodicTimer("TICK", Tick.INSTANCE, TICK_INTERVAL);
+    }
+
+    /**
+     * Computes the time we should schedule an extra tick for, if we'll need an extra tick before the next naturally occurring one.
+     *
+     * @param now The current time.
+     * @param timeToAwaken The time we want to wake up right after.
+     * @return The time until we should wake up, if that's before the next tick; else {@code empty}.
+     */
+    protected static Optional<FiniteDuration> timeUntilExtraTick(final Instant now, final Instant timeToAwaken) {
+        final FiniteDuration delta = Duration.fromNanos(ChronoUnit.NANOS.between(now, timeToAwaken));
+        if (delta.lt(TICK_INTERVAL)) {
+            return Optional.of(delta.plus(SLEEP_SLOP));
```

Comment:
Can you comment on the point of `SLEEP_SLOP`? Are we concerned that sleeping exactly to the time will leave us just short? Usually timers sleep for at least the requested amount of time and are prone to sleeping longer when very short times are requested. The worst thing that happens is we iterate a lot of times until the interval gets so small that the timer resolution cannot sleep for that little. Perhaps a counter metric here without slop is in order; if we see spikes in the sum of the counter then we'll know that we've got a job that's stuck in Zeno's paradox.

Reply:
Yeah, upon reflection this doesn't seem necessary! I've removed `SLEEP_SLOP`.

Is this a reasonable approach to log a counter? Haven't yet committed.
<img width="1704" alt="screenshot 2019-01-15 14 03 17" src="https://user-images.githubusercontent.com/1899701/51213009-cb9e7580-18ce-11e9-97c7-63d3dd09c885.png">

Reply:
I'm now [using `PeriodicMetrics` instead](https://github.com/ArpNetworking/metrics-portal/pull/144/files#diff-0cd7c6e23360f8f50d83def6cc256213R82), as you suggested. Seems much better.